test/e2e/utils_test.go

```diff
@@ -180,6 +210,45 @@ func patchControllerImage(baseManifestPath string, image string, tag string) (io
 	return bytes.NewReader(kustomized), nil
 }
 
+const kongImageKustomizationContents = `resources:
+- base.yaml
+images:
+- name: kong
+  newName: %v
+  newTag: '%v'
+- name: kong/kong-gateway
+  newName: %v
+  newTag: '%v'
+`
+
+func patchKongImage(baseManifestsReader io.Reader, image string, tag string) (io.Reader, error) {
+	workDir, err := os.MkdirTemp("", "kictest.")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(workDir)
+	orig, err := io.ReadAll(baseManifestsReader)
+	if err != nil {
+		return nil, err
+	}
+
+	err = os.WriteFile(filepath.Join(workDir, "base.yaml"), orig, 0o600)
+	if err != nil {
+		return nil, err
+	}
+
+	kustomization := []byte(fmt.Sprintf(kongImageKustomizationContents, image, tag, image, tag))
+	err = os.WriteFile(filepath.Join(workDir, "kustomization.yaml"), kustomization, 0o600)
+	if err != nil {
+		return nil, err
+	}
+	kustomized, err := kustomizeManifest(workDir)
```

Comment:
Just a thought: have you looked at https://pkg.go.dev/sigs.k8s.io/kustomize/kyaml@v0.13.9/filesys#MakeFsInMemory? It seems that using it should be as simple as swapping it out in places where `filesys.FileSystem` is used, and we get the benefit of not littering the local filesystem and not needing to clean up after ourselves.

Reply:
I think we could do this in another issue and PR.

Reply:
Do you mind creating an issue for this?

Reply:
https://github.com/Kong/kubernetes-ingress-controller/issues/2896